core/state/snapshot/journal.go

```diff
@@ -108,44 +108,15 @@ func loadAndParseJournal(db ethdb.KeyValueStore, base *diskLayer) (snapshot, jou
 	// So if there is no journal, or the journal is invalid(e.g. the journal
 	// is not matched with disk layer; or the it's the legacy-format journal,
 	// etc.), we just discard all diffs and try to recover them later.
-	journal := rawdb.ReadSnapshotJournal(db)
-	if len(journal) == 0 {
-		log.Warn("Loaded snapshot journal", "diskroot", base.root, "diffs", "missing")
-		return base, generator, nil
-	}
-	r := rlp.NewStream(bytes.NewReader(journal), 0)
-
-	// Firstly, resolve the first element as the journal version
-	version, err := r.Uint()
+	var current snapshot = base
+	err := iterateJournal(db, func(parent common.Hash, root common.Hash, destructSet map[common.Hash]struct{}, accountData map[common.Hash][]byte, storageData map[common.Hash]map[common.Hash][]byte) error {
```

Comment:
I have to point out that new code has slightly changed the behavior. But my opinion is new code is better.

Previously it's a bit messy, if 
(1) journal version it not resolved, 
(2) journal version is not matched 
(3) journal root hash is not matched, 
we will throw out the journal and only use disk layer

if (1) we failed to resolve root hash in journal
(2) we failed to load diff layer from journal
we just try to rebuild the entire snapshot even the disk layer is correct.

But now, if there is anything wrong in journal, we just throw it away and use diff layer only. It makes more sense to me.